cadquery/cq.py

```diff
@@ -1046,6 +1051,43 @@ def compounds(
         """
         return self._selectObjects("Compounds", selector, tag)
 
+    def ancestors(self: T, kind: Shapes, tag: Optional[str] = None) -> T:
+        """
+        Iterate over ancestors.
+
+        :param kind: kind of ancestor, e.g. "Face" or "Edge"
+        :param tag: if set, search the tagged object instead of self
+        :return: a Workplane object who's stack contains all ancestors.
```

Comment:
I'll do some testing today.  First a couple of docstring grammar/typo issues.

"who's" should be "whose".  I see this form comes from other Workplane docstrings.

```
$ find ./cadquery -name "*.py" -exec grep -Hn "who's" {} \;
./cadquery/cq.py:865:        :return: a CQ object who's stack contains  the *distinct* vertices of *all* objects on the
./cadquery/cq.py:899:        :return: a CQ object who's stack contains all of the *distinct* faces of *all* objects on
./cadquery/cq.py:934:        :return: a CQ object who's stack contains all of the *distinct* edges of *all* objects on
./cadquery/cq.py:968:        :return: a CQ object who's stack contains all of the *distinct* wires of *all* objects on
./cadquery/cq.py:994:        :return: a CQ object who's stack contains all of the *distinct* solids of *all* objects on
./cadquery/cq.py:1023:        :return: a CQ object who's stack contains all of the *distinct* shells of *all* objects on
./cadquery/cq.py:1046:        :return: a CQ object who's stack contains all of the *distinct* compounds of *all* objects on
./cadquery/cq.py:1060:        :return: a Workplane object who's stack contains all ancestors.
./cadquery/cq.py:1080:        :return: a Workplane object who's stack contains all ancestors.
```

Reply:
Fixed